compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -2446,10 +2507,20 @@ fn add_upstream_rust_crates<'a>(
                             | NativeLibKind::Framework { .. }
                             | NativeLibKind::Unspecified
                             | NativeLibKind::RawDylib => {}
-                            NativeLibKind::Static {
-                                bundle: Some(true) | None,
-                                whole_archive: _,
-                            } => {}
+                            NativeLibKind::Static { bundle: Some(true) | None, whole_archive } => {
+                                if sess.opts.unstable_opts.separate_native_rlib_dependencies {
+                                    // if rlib contains native libs as archieves
+                                    // they unpacked to tmpdir
+                                    let lib = name.to_string();
+                                    let lib =
+                                        String::from("lib") + &lib + String::from(".a").as_ref();
```

Comment:
```suggestion
                                    let lib = format!("lib{name}.a");
```

Reply:
Also `sess.target.staticlib_prefix` and `sess.target.staticlib_suffix` need to be used instead of "lib" and ".a", and `verbatim` flag should also be considered (here and in other places).
See e.g. `fn find_library` for an example.